From ca85cc9d278d19eef92d9ed37668e906522f56c2 Mon Sep 17 00:00:00 2001
From: Artem Iglikov <artikz@google.com>
Date: Tue, 12 Jun 2018 11:40:24 +0000
Subject: [PATCH 04/71] DO NOT MERGE Revert "Fix gray out previously connected
 device preference issue"

This reverts commit 95abf59af43bafe4922b8cc096b10b65440419b8.

Reason for revert: Crashes settings (b/110075201)

Bug: 110075201
Change-Id: Ie6ea94e0891754f18576bd21447b64410d2fd23a
---
 ...ouslyConnectedDevicePreferenceController.java | 16 +++-------------
 ...yConnectedDevicePreferenceControllerTest.java |  6 ------
 2 files changed, 3 insertions(+), 19 deletions(-)

diff --git a/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceController.java b/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceController.java
index 73c95ab4e3..ba3ee1f105 100644
--- a/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceController.java
+++ b/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceController.java
@@ -21,29 +21,26 @@ import android.support.annotation.VisibleForTesting;
 import android.support.v7.preference.Preference;
 import android.support.v7.preference.PreferenceScreen;
 
+import android.util.Log;
 import com.android.settings.bluetooth.BluetoothDeviceUpdater;
 import com.android.settings.bluetooth.SavedBluetoothDeviceUpdater;
-import com.android.settings.connecteddevice.dock.DockUpdater;
 import com.android.settings.core.BasePreferenceController;
 import com.android.settings.dashboard.DashboardFragment;
-import com.android.settings.overlay.FeatureFactory;
+
 import com.android.settingslib.core.lifecycle.LifecycleObserver;
 import com.android.settingslib.core.lifecycle.events.OnStart;
 import com.android.settingslib.core.lifecycle.events.OnStop;
+import com.android.settingslib.utils.ThreadUtils;
 
 public class PreviouslyConnectedDevicePreferenceController extends BasePreferenceController
         implements LifecycleObserver, OnStart, OnStop, DevicePreferenceCallback {
 
     private Preference mPreference;
     private BluetoothDeviceUpdater mBluetoothDeviceUpdater;
-    private DockUpdater mSavedDockUpdater;
     private int mPreferenceSize;
 
     public PreviouslyConnectedDevicePreferenceController(Context context, String preferenceKey) {
         super(context, preferenceKey);
-
-        mSavedDockUpdater = FeatureFactory.getFactory(
-                context).getDockUpdaterFeatureProvider().getSavedDockUpdater(context, this);
     }
 
     @Override
@@ -65,14 +62,12 @@ public class PreviouslyConnectedDevicePreferenceController extends BasePreferenc
     @Override
     public void onStart() {
         mBluetoothDeviceUpdater.registerCallback();
-        mSavedDockUpdater.registerCallback();
         updatePreferenceOnSizeChanged();
     }
 
     @Override
     public void onStop() {
         mBluetoothDeviceUpdater.unregisterCallback();
-        mSavedDockUpdater.unregisterCallback();
     }
 
     public void init(DashboardFragment fragment) {
@@ -97,11 +92,6 @@ public class PreviouslyConnectedDevicePreferenceController extends BasePreferenc
         mBluetoothDeviceUpdater = bluetoothDeviceUpdater;
     }
 
-    @VisibleForTesting
-    void setSavedDockUpdater(DockUpdater savedDockUpdater) {
-        mSavedDockUpdater = savedDockUpdater;
-    }
-
     @VisibleForTesting
     void setPreferenceSize(int size) {
         mPreferenceSize = size;
diff --git a/tests/robotests/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceControllerTest.java b/tests/robotests/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceControllerTest.java
index d04b206c6d..444728a1c4 100644
--- a/tests/robotests/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceControllerTest.java
+++ b/tests/robotests/src/com/android/settings/connecteddevice/PreviouslyConnectedDevicePreferenceControllerTest.java
@@ -21,7 +21,6 @@ import android.content.pm.PackageManager;
 import android.support.v7.preference.Preference;
 
 import com.android.settings.bluetooth.BluetoothDeviceUpdater;
-import com.android.settings.connecteddevice.dock.DockUpdater;
 import com.android.settings.dashboard.DashboardFragment;
 import com.android.settings.testutils.SettingsRobolectricTestRunner;
 
@@ -47,8 +46,6 @@ public class PreviouslyConnectedDevicePreferenceControllerTest {
     @Mock
     private BluetoothDeviceUpdater mBluetoothDeviceUpdater;
     @Mock
-    private DockUpdater mDockUpdater;
-    @Mock
     private PackageManager mPackageManager;
 
     private Context mContext;
@@ -64,7 +61,6 @@ public class PreviouslyConnectedDevicePreferenceControllerTest {
         mPreConnectedDeviceController =
                 new PreviouslyConnectedDevicePreferenceController(mContext, KEY);
         mPreConnectedDeviceController.setBluetoothDeviceUpdater(mBluetoothDeviceUpdater);
-        mPreConnectedDeviceController.setSavedDockUpdater(mDockUpdater);
 
         mPreference = new Preference(mContext);
         mPreConnectedDeviceController.setPreference(mPreference);
@@ -75,12 +71,10 @@ public class PreviouslyConnectedDevicePreferenceControllerTest {
         // register the callback in onStart()
         mPreConnectedDeviceController.onStart();
         verify(mBluetoothDeviceUpdater).registerCallback();
-        verify(mDockUpdater).registerCallback();
 
         // unregister the callback in onStop()
         mPreConnectedDeviceController.onStop();
         verify(mBluetoothDeviceUpdater).unregisterCallback();
-        verify(mDockUpdater).unregisterCallback();
     }
 
     @Test
-- 
2.17.1

